### DISCUSSION TITLE
disable password login

### DISCUSSION THREAD

**Discussion by whyameye** [shellhub-io/shellhub#2966] [Q&A]
I am self-hosting and have public key login working for a specified device. I ssh from the CLI now using the SSHID and no longer need to enter a password. However, if I do not provide the certificate the agent still allows me to login using my password. Is there a way I can now disable password login? I have disabled password login on the agent's ssh server but it appears that is not enough.

**Comment by gustavosbarreto** [comment]
> I have disabled password login on the agent's ssh server but it appears that is not enough.

The ShellHub Agent doesn't use the standard SSH server, so if you disable password login, it will not take effect.

**Comment by whyameye** [comment]
you didn't offer a solution but I found a workaround for now which is to edit the password method in the agent to just return false.
